### PR TITLE
BUG: Categorical setitem, comparison with tuple category

### LINKED ISSUE #20439: Categorical.__setitem__ doesn't handle setting tuples
We allow tuples in the categories, since they're hashable. But `Categorical.__setitem__` has an incorrect check before setting.

```python
In [3]: c = pd.Categorical([(0, 1), (1, 2)])

In [5]: c[0] = (1, 2)
```

Raises

```pytb
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-5-4d3f2e8a3cfd> in <module>()
----> 1 c[0] = (1, 2)

/Users/taugspurger/miniconda3/envs/pandas-0.19.2/lib/python3.5/site-packages/pandas/core/categorical.py in __setitem__(self, key, value)
   1652         # something to np.nan
   1653         if len(to_add) and not isnull(to_add).all():
-> 1654             raise ValueError("Cannot setitem on a Categorical with a new "
   1655                              "category, set the categories first")
   1656

ValueError: Cannot setitem on a Categorical with a new category, set the categories first
```

If it's easy, this is worth supporting. If it's complicated, then maybe not.

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #20439
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Review by WillAyd — COMMENTED** [review]
lgtm - can you add a whatsnew?

**Review by WillAyd — APPROVED** [review]
lgtm @TomAugspurger

File: pandas/core/arrays/categorical.py [hunk 1]
```diff
@@ -1883,7 +1885,8 @@ def _validate_setitem_value(self, value):
             new_codes = self._validate_listlike(value)
             value = Categorical.from_codes(new_codes, dtype=self.dtype)
 
-        rvalue = value if is_list_like(value) else [value]
+        # wrap scalars and hashable-listlikes in list
+        rvalue = value if not is_hashable(value) else [value]
```
Comment (jreback) [inline pandas/core/arrays/categorical.py:1889]: can use `maybe_make_list` (but certainly can followup as may need to change that definition slightly)

**Review by jreback — CHANGES_REQUESTED** [review]
can you also check getitem?

**Comment by jbrockmendel** [comment]
> can you also check getitem?

not sure i follow.  you mean check that `cat[i]` gives a tuple.  pretty sure we already have tests for that

**Comment by jreback** [comment]
> > can you also check getitem?
> 
> not sure i follow. you mean check that `cat[i]` gives a tuple? pretty sure we already have tests for that

ok then.

**Review by jreback — APPROVED** [review]

**Pull request merged**